### PR TITLE
[DOC] RDoc for Open3

### PR DESCRIPTION
**Pull request by BurdetteLamar**

### CODE REVIEW

File: lib/open3.rb [hunk 1]
```diff
@@ -960,7 +960,7 @@ def pipeline_rw(*cmds, &block)
   module_function :pipeline_rw
 
   # :call-seq:
-  #   Open3.pipeline_r([env, ] *cmds, options = {}) -> [last_stdout, wait_threads]
+  #   Open3.pipeline_r([env, ] *cmds, options = {}) -> array_of_statuses
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply lib/open3.rb:963]: Nope.  All fixed now, I hope.

File: lib/open3.rb [hunk 2]
```diff
@@ -1030,33 +1034,71 @@ def pipeline_r(*cmds, &block)
...
   module_function :pipeline_r
 
-  # Open3.pipeline_w starts a list of commands as a pipeline with a pipe
-  # which connects to stdin of the first command.
-  #
-  #   Open3.pipeline_w(cmd1, cmd2, ... [, opts]) {|first_stdin, wait_threads|
-  #     ...
-  #   }
+  # :call-seq:
+  #   Open3.pipeline_w([env, ] *cmds, options = {}) -> array_of_statuses
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply lib/open3.rb]: Fixed.

File: lib/open3.rb [hunk 3]
```diff
@@ -1030,33 +1034,71 @@ def pipeline_r(*cmds, &block)
...
+  #
+  # Unlike Process.spawn, this method waits for the child processes to exit
+  # before returning, so the caller need not do so.
+  #
+  # If the first argument is a hash, it becomes leading argument +env+
+  # in each call to Process.spawn;
+  # see {Execution Environment}[https://docs.ruby-lang.org/en/master/Process.html#module-Process-label-Execution+Environment].
+  #
+  # If the last argument is a hash, it becomes trailing argument +options+
+  # in each call to Process.spawn'
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply lib/open3.rb]: Fixed.

File: lib/open3.rb (hunk 1 above)
Comment (peterzhu2118) [inline lib/open3.rb:963]:
```suggestion
  #   Open3.pipeline_r([env, ] *cmds, options = {}) -> [last_stdout, wait_threads]
```

**Review by peterzhu2118 — APPROVED** [review]

**Pull request merged**